adapters/appnexus/appnexus.go

```diff
@@ -433,26 +439,33 @@ func (a *AppNexusAdapter) MakeBids(internalRequest *openrtb.BidRequest, external
 	for _, sb := range bidResp.SeatBid {
 		for i := 0; i < len(sb.Bid); i++ {
 			bid := sb.Bid[i]
-			if bidType, err := getMediaTypeForBid(&bid); err == nil {
-				bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
-					Bid:     &bid,
-					BidType: bidType,
-				})
-			} else {
+			var impExt appnexusBidExt
+			if err := json.Unmarshal(bid.Ext, &impExt); err != nil {
 				errs = append(errs, err)
+			} else {
+				if bidType, err := getMediaTypeForBid(&impExt); err == nil {
```

Comment:
Not sure I understand this logic here. 
Do we need to set IabCategory ONLY if bidType exists? No Checks if bid type is "video"? Why `if len(bid.Cat) == 0` is inside the `if bidType, err := getMediaTypeForBid(&impExt); err == nil`

Reply:
Sure, so getMediaTypeForBid() was a pre-existing function the adapter uses to map an appnexus bid type value back to an OpenRTB bid type value. If that function failed the code before adds an error for the bid instead of adding a bid response. If it doesn't error then I'm adding the call to map an AppNexus category to an IAB category and storing the result in the bid.Cat field, but I only want to  do this if impbus did not already return an IAB category in bid.Cat (a.k.a. only if len(bid.Cat) == 0). Maybe we only want to map an AppNexus bid to an IAB bid for video bids, I didn't put that check in yet though, do you guys think setting an IAB category in bid.Cat is something we should only do for videos?

Reply:
Thank you, I got the logic about nested if-statements, it makes sense.